src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,116 @@
+package duke;
+
+import java.text.ParseException;
+
+public class Parser {
+    // TODO: 20/8/20 CHANGE ALL MATCHES TO matcher.find
+    public static Command parseCommand(String echo, TaskList tasks) throws DukeException, ParseException {
+        if (echo.equals("bye")) {
+            return new ExitCommand();
+        } else if (echo.matches("(?i)list\\s*")) { // Querying items
+            return new ListCommand();
+        } else if (echo.matches("(?i)find\\s+\\S+")) {
+            return new FindCommand(echo);
+        } else if (echo.matches("(?i)^done.*")) {  // Checks if it matches done and an integer
+            int index = Parser.parseDone(echo, tasks);
+            return new DoneCommand(index);
+        } else if (echo.matches("(?i)^delete.*")) { // Checks if it matches delete and an integer
+            int index = Parser.parseDelete(echo, tasks.length());
+            return new DeleteCommand(index);
+        } else if (echo.matches("(?i)^todo.*")) {
+            Task toAdd = Parser.parseToDo(echo);
+            return new ToDoCommand(toAdd);
+        } else if (echo.matches("(?i)deadline.*")) {
+            Task toAdd = Parser.parseDeadline(echo);
+            return new DeadlineCommand(toAdd);
+        } else if (echo.matches("(?i)event.*")) {
+            Task toAdd = Parser.parseEvent(echo);
+            return new EventCommand(toAdd);
+        } else {
+            throw new DukeException("Invalid command.");
+        }
+    }
+
+    public static Task parseToDo(String echo) throws DukeException {
+        if (echo.matches("(?i)^todo\\s+\\S+.*")) {
+            String text = echo.replaceFirst("(?i)^todo\\s*", "");
+            return new ToDo(text);
+        } else if (echo.matches("(?i)^todo\\s*")) {
+            throw new DukeException("Please put a description to your task.");
+        } else {
+            throw new DukeException("Please put a space between the command and description.");
+        }
+    }
+
+    public static Task parseDeadline(String echo) throws DukeException {
+        if (echo.matches("(?i)^deadline\\s+\\S+.*\\s+\\/by\\s+\\S+.*")) {
```

Comment:
Perhaps can simplify this or use another way to express it? Looks very complicated.

Reply:
Unfortunately no, the regex used forces the following:

`(?i)`: case insensitive matching
`^`: at the start of the command,
`deadline`: match the word `deadline` exactly
`\\s+`: followed by at least one space character
`\\S+`: followed by at least one non-space character (i.e. the description)
`.*`: followed by anything (to support for multi-word descriptions)
`\\s+`: followed by at least one space character
`\\/`: followed by the character `/`
`by`: followed by `by`
`\\s+`: followed by at least one space character
`\\S+`: followed by at least one non-space character (i.e. there is a time)
`.*`: followed by anything (to support for multi-word descriptions)